app/main/views/index.py

```diff
@@ -184,6 +184,12 @@ def download_document(service_id, document_id):
 
 def _format_file_expiry_date(expiry_date_as_str: str) -> str:
     file_expiry_date = (parser.parse(expiry_date_as_str)).date()
+
+    # if expiry date passed, even if file is still available, we do not return it to respect data retention period
+    # set by the service
+    if file_expiry_date - date.today() <= timedelta(days=-1):
+        abort(404)
```

Comment:
I think throwing the 404 from this helper function is a bit funky - can we clean this up somehow to have this function just handle formatting the date correctly, and move the 404 logic back to the view?

This function is called from within the final line of the view where we render_template - seems maybe slightly surprising that we're throwing a 404 from there.